src/orders/components/OrderRefundDatagrid/OrderRefundDatagrid.tsx

```diff
@@ -76,39 +80,40 @@ export const OrderRefundDatagrid: React.FC<OrderRefundDatagridProps> = ({
         <Text size={5} fontWeight="bold">
           <FormattedMessage {...refundGridMessages.refundSection} />
         </Text>
-        {/** TODO: Add modal */}
-        <Button variant="secondary">
+        <Button variant="secondary" onClick={onRefundAdd}>
           <PlusIcon />
           <FormattedMessage {...refundGridMessages.addNewRefund} />
         </Button>
       </Box>
-      <DatagridChangeStateContext.Provider value={datagrid}>
-        <Datagrid
-          readonly
-          hasRowHover
-          rowMarkers="none"
-          columnSelect="none"
-          freezeColumns={2}
-          menuItems={getMenuItems}
-          verticalBorder={col => col > 1}
-          availableColumns={visibleColumns}
-          emptyText={""}
-          getCellContent={getCellContent}
-          getCellError={() => false}
-          rows={grantedRefunds.length}
-          selectionActions={() => null}
-          onColumnResize={handlers.onResize}
-          onColumnMoved={handlers.onMove}
-          recentlyAddedColumn={recentlyAddedColumn}
-          renderColumnPicker={() => (
-            <ColumnPicker
-              selectedColumns={selectedColumns}
-              staticColumns={staticColumns}
-              onToggle={handlers.onToggle}
-            />
-          )}
-        />
-      </DatagridChangeStateContext.Provider>
+      {grantedRefunds.length !== 0 && (
```

Comment:
question: don't we want to show some empty state?

Reply:
We have no designs for this case, but I will try to come up with something